src/orange/surf/detail/SurfaceAction.hh

```diff
@@ -76,7 +76,7 @@ struct StaticSurfaceAction
 
 #define ORANGE_SURF_DISPATCH_CASE_IMPL(FUNC, TYPE) \
     case SurfaceType::TYPE:                        \
-        FUNC(typename SurfaceTypeTraits<SurfaceType::TYPE>::type) break
+        FUNC(typename SurfaceTypeTraits<SurfaceType::TYPE>::type)
```

Comment:
Consider adding somewhere that FUNC (maybe in a note for ORANGE_SURF_DISPATCH_IMPL or by renaming) must be a `return` statement (of a function call) (rather than 'just' a function call)

Reply:
Added detailed documentation to this implementation detail